src/AzureAppConfigurationImpl.ts

```diff
@@ -98,39 +105,38 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
 
         if (options?.refreshOptions?.enabled) {
-            const { watchedSettings, refreshIntervalInMs } = options.refreshOptions;
-            // validate watched settings
+            const { refreshIntervalInMs, watchedSettings } = options.refreshOptions;
             if (watchedSettings === undefined || watchedSettings.length === 0) {
-                throw new Error("Refresh is enabled but no watched settings are specified.");
+                this.#watchAll = true; // if no watched settings is specified, then watch all
+            } else {
+                for (const setting of watchedSettings) {
+                    if (setting.key.includes("*") || setting.key.includes(",")) {
+                        throw new Error("The characters '*' and ',' are not supported in key of watched settings.");
+                    }
+                    if (setting.label?.includes("*") || setting.label?.includes(",")) {
+                        throw new Error("The characters '*' and ',' are not supported in label of watched settings.");
+                    }
+                    this.#sentinels.push(setting);
+                }
             }
 
             // custom refresh interval
             if (refreshIntervalInMs !== undefined) {
                 if (refreshIntervalInMs < MIN_REFRESH_INTERVAL_IN_MS) {
                     throw new Error(`The refresh interval cannot be less than ${MIN_REFRESH_INTERVAL_IN_MS} milliseconds.`);
-
                 } else {
-                    this.#refreshInterval = refreshIntervalInMs;
+                    this.#kvRefreshInterval = refreshIntervalInMs;
                 }
             }
-
-            for (const setting of watchedSettings) {
-                if (setting.key.includes("*") || setting.key.includes(",")) {
-                    throw new Error("The characters '*' and ',' are not supported in key of watched settings.");
-                }
-                if (setting.label?.includes("*") || setting.label?.includes(",")) {
-                    throw new Error("The characters '*' and ',' are not supported in label of watched settings.");
-                }
-                this.#sentinels.push(setting);
-            }
-
-            this.#refreshTimer = new RefreshTimer(this.#refreshInterval);
+            this.#kvRefreshTimer = new RefreshTimer(this.#kvRefreshInterval);
         }
 
+        this.#kvSelectors = getValidKeyValueSelectors(options?.selectors);

```

Comment:
Why does this KV selector happen outside the `if (options?.refreshOptions?.enabled) {` while the FF selector happens inside it's `if`?

Reply:
Because we will always have key value selector, if user doesn't list any, we will use default selecter i.e. keyfilter="*", labelFilter=null. But for feature flag, only when user enabled feature flag, we will load it into configuration. This behavior is the same as .NET provider where you need to call `UseFeatureFlag`.

BTW, `if (options?.refreshOptions?.enabled)` branch checks the refresh option. I moved `this.#kvSelectors = getValidKeyValueSelectors(options?.selectors);` to the front of `if (options?.refreshOptions?.enabled)` and add more comments to make it more readable.